### PR TITLE
Added validation on create pop-up, search bar inside org details

### PR DESCRIPTION
**Pull request by himanshi-chhabra**
Signed-off-by: Himanshi Chhabra <hchhabra@progress.com>

### :nut_and_bolt: Description: What code changed, and why?
- Add validation on name field for the create pop-up of Role/ Data bag / Data bag item/ Client/ Environment Tabs.
- Add validation on search box for  Role/ Data bag item/ Client/ Environment Tabs.
- Add unit testcase for Validate of Role/ Data bag item/ Client/ Environment Tabs.
- Add validation on Run list search bar

### :chains: Related Resources
#4960 
### :+1: Definition of Done
 **VALDATION:-**   Added validation that name can only contain ```alphabets``` **a-z, A-Z**, ```numbers``` **0-9**, ```Underscore```  **_**, ```Hyphen```  **-**.
- Added this above validation on  name/id field for the create pop-up of Role/ Data bag / Data bag item/ Client/ Environment Tabs.
- Added this above validation on search box for  Role/ Data bag item/ Client/ Environment Tabs.
- Added unit testcase for above validation that form is valid or invalid of Role/ Data bag item/ Client/ Environment Tabs.
- Added the above validation on the Run list search bar.
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab
```
To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
Steps to reproduce:
   - Go To  Infrastructure -> Chef Servers -> click server name-> click on org name ->
  -  **Steps for Validate Create Role:** 
           1. Click Roles -> Create Role -> add name with ```test role``` -> show error -> ```name can only contain 'a-z', 'A-Z', '0-9', '_', '-'.``` and create button is ```disabled```.
           2. add name with ```test_role-1:::::``` -> show error -> ```name can only contain 'a-z', 'A-Z', '0-9', '_', '-'.``` and create button is ```disabled```.
           3. add name with ```test_role-1``` -> no error.
           4. Same for others tab.
  -  **Steps for Validate Search Role:** 
         1. Click Roles -> add text on search bar with ```test@role``` -> show error -> ```name can only contain 'a-z', 'A-Z', '0-9', '_', '-'.``` and search button is ```disabled```.
         2. add name with ```test[role]-1:::::``` -> ```no roles available with this name.```
         3. add name with ```test_role-1``` -> no error.
         4. Same for others tab. 
   -  **Steps for Validate Run list search bar:** 
         1. Click Roles -> click Create Role -> add name and description -> click runlist
         2.  add text on search bar with ```role(role-1``` -> show error -> ```name can only contain 'a-z', 'A-Z', '0-9', '_', '-'.```
         3. add name with ```test[role]-1``` -> show ```Not available```.
         4. add name with ```test_role-1``` -> no error.

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/61003053/117340823-f73e3300-aebe-11eb-8289-5561d51a5332.mp4

- For run list search:
Chef-Automate (20)

### CODE REVIEW

**Review by vinay033 — APPROVED** [review]
nice work @himanshi-chhabra 🎉

**Comment by chaitali-mane** [comment]
When I placed # at first place in textbox, will not throw error. But in search box, show error. Please find below recording.

https://user-images.githubusercontent.com/71449322/117103467-0d9d9f00-ad98-11eb-831b-44d12fc05ebb.mp4

File: components/automate-ui/src/app/modules/infra-proxy/create-client-modal/create-client-modal.component.html [hunk 1]
```diff
@@ -6,9 +6,13 @@ <h2 slot="title">Create Client</h2>
         <chef-form-field>
           <label *ngIf="!created">
             <span class="label">Client Name</span>
-            <input chefInput firstFocus formControlName="name" type="text" (keyup)="handleNameInput($event)" autocomplete="off" />
+            <input chefInput firstFocus formControlName="name" type="text" (keyup)="handleInput($event)" autocomplete="off" />
           </label>
           <chef-error *ngIf="error">{{error}}</chef-error>
+          <chef-error
+            *ngIf="createForm.get('name').hasError('pattern') && createForm.get('name').dirty">
+            name can only contain 'a-z', 'A-Z', '0-9', '_', '-'.
```
Comment (chaitali-mane) [inline components/automate-ui/src/app/modules/infra-proxy/create-client-modal/create-client-modal.component.html:14]: Other error message starts with capitalise, this error not.

**Comment by himanshi-chhabra** [comment]
> When I placed # at first place in textbox, will not throw error. But in search box, show error. Please find below recording.
> 
>  Validation.mp4

@chaitali-mane  Update the changes

**Comment by chaitali-mane** [comment]
When put @ at first place , it's not show error but create button is disabled. 
@issue-validation

**Comment by himanshi-chhabra** [comment]
> When put @ at first place , it's not show error but create button is disabled.
> @issue-validation

updated the changes.

**Review by chaitali-mane — APPROVED** [review]
Changes looks good to me , thanks @himanshi-chhabra

**Pull request merged**